NitroxModel/Discovery/InstallationFinders/SteamFinder.cs

```diff
@@ -0,0 +1,165 @@
+            if (string.IsNullOrWhiteSpace(homePath))
+            {
+                homePath = Environment.GetEnvironmentVariable("HOME");
+            }
+
+            if (!Directory.Exists(homePath))
+            {
+                return null;
+            }
+
+            string[] commonSteamPath =
+            [
+                // Default install location
+                // https://github.com/ValveSoftware/steam-for-linux
+                Path.Combine(homePath, ".local", "share", "Steam"),
+                // Those symlinks are often use as a backward-compatibility (Debian, Ubuntu, Fedora, ArchLinux)
+                // https://wiki.archlinux.org/title/steam, https://askubuntu.com/questions/227502/where-are-steam-games-installed
+                Path.Combine(homePath, ".steam", "steam"),
+                Path.Combine(homePath, ".steam", "root"),
+                // Flatpack install
+                // https://github.com/flathub/com.valvesoftware.Steam/wiki, https://flathub.org/apps/com.valvesoftware.Steam
+                Path.Combine(homePath, ".var", "app", "com.valvesoftware.Steam", ".local", "share", "Steam"),
+                Path.Combine(homePath, ".var", "app", "com.valvesoftware.Steam", ".steam", "steam"),
+            ];
+
+            foreach (string path in commonSteamPath)
+            {
+                if (Directory.Exists(path))
+                {
+                    return path;
+                }
+            }
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            string homePath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+            if (string.IsNullOrWhiteSpace(homePath))
+            {
+                homePath = Environment.GetEnvironmentVariable("HOME");
+            }
+
+            if (!Directory.Exists(homePath))
+            {
+                return null;
+            }
+
+            // Steam should always be here
+            string steamPath = Path.Combine(homePath, "Library", "Application Support", "Steam");
+            if (Directory.Exists(steamPath))
+            {
+                return steamPath;
+            }
+        }
+
+        return null;
+    }
+
+    /// <summary>
+    /// Finds game install directory by iterating through all the steam game libraries configured, matching the given appid.
+    /// </summary>
+    private static string SearchAllInstallations(string libraryFolders, int appid, string gameName)
+    {
+        if (!File.Exists(libraryFolders))
+        {
+            return null;
+        }
+
+        StreamReader file = new(libraryFolders);
+        char[] trimChars = [' ', '\t'];
+
+        while (file.ReadLine() is { } line)
+        {
+            line = Regex.Unescape(line.Trim(trimChars));
+            Match regMatch = Regex.Match(line, "\"(.*)\"\t*\"(.*)\"");
+            string key = regMatch.Groups[1].Value;
```

Comment:
Do we need to check that a match has been found first or this will throw a null pointer potentially on the index